src/main/java/seedu/address/model/Model.java

```diff
@@ -77,6 +77,30 @@
      */
     void setPerson(Person target, Person editedPerson);
 
+    /**
+     * Returns true if an event with the same identity as {@code event} exists in the address book.
+     */
+    boolean hasEvent(Event event);
+
+    /**
+     * Deletes the given event.
+     * The event must exist in the address book.
+     */
+    void deleteEvent(Event event);
+
+    /**
+     * Adds the given event.
+     * {@code event} must not already exist in the address book.
+     */
+    void addEvent(Event event);
+
+    /**
+     * Replaces the given event {@code target} with {@code editedEvent}.
+     * {@code target} must exist in the address book.
+     * The event identity of {@code editedPerson} must not be the same as another existing event in the address book.
+     */
+    void setEvent(Event target, Event editedEvent);
```

Comment:
Changes made here.

Reply:
Is there a typo in the javadocs? `editedEvent` instead of `editedPerson`

Reply:
Yeah forgot to change it. Thanks